src/js/actions/transform.js

```diff
@@ -1023,14 +1023,19 @@ define(function (require, exports) {
         _layerTransformHandler = synchronization.debounce(function (event) {
             this.dispatch(events.ui.TOGGLE_OVERLAYS, { enabled: true });
             
+            var appStore = this.flux.store("application"),
+                currentDoc = appStore.getCurrentDocument(),
+                artboardsSelected = currentDoc.layers.selected.some(function (layer) {
```

Comment:
I think `hasArtboardSelected` would be a good candidate for a layerstructure prop.  You've already used it twice in this PR!  (future suggestion)